driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java

```diff
@@ -238,6 +249,10 @@ private OIndexManager getIndexManager() {
 
     @Override
     public Iterator<Edge> edges(Object... edgeIds) {
+        Set<Class<?>> edgeIdTypes = Stream.of(edgeIds).map(edgeId -> edgeId.getClass()).collect(toSet());
+        if(edgeIdTypes.size() > 1 && edgeIdTypes.contains(OrientEdge.class))
+            throw Graph.Exceptions.idArgsMustBeEitherIdOrElement();
```

Comment:
@mpollmeier 
whould like to hears your thoughts on this....

the structure tests expect this validation